Src/ILGPU/IR/Values/LanguageValues.cs

```diff
@@ -0,0 +1,216 @@
+    ILGPU.Util.FormatString.FormatExpression>;
+using ValueList = ILGPU.Util.InlineList<ILGPU.IR.Values.ValueReference>;
+
+namespace ILGPU.IR.Values
+{
+    /// <summary>
+    /// Represents the kind of inline language.
+    /// </summary>
+    public enum LanguageKind
+    {
+        /// <summary>
+        /// Inline PTX assembly.
+        /// </summary>
+        PTX,
+    }
+
+    /// <summary>
+    /// Represents the <see cref="Accelerator.LanguageVersion"/> property.
+    /// </summary>
+    [ValueKind(ValueKind.LanguageVersion)]
+    public sealed class LanguageVersionValue : DeviceConstantValue
+    {
+        #region Instance
+
+        /// <summary>
+        /// Constructs a new value.
+        /// </summary>
+        /// <param name="initializer">The value initializer.</param>
+        internal LanguageVersionValue(in ValueInitializer initializer)
+            : base(
+                  initializer,
+                  initializer.Context.GetPrimitiveType(BasicValueType.Int32))
+        { }
+
+        #endregion
+
+        #region Properties
+
+        /// <summary cref="Value.ValueKind"/>
+        public override ValueKind ValueKind => ValueKind.LanguageVersion;
+
+        #endregion
+
+        #region Methods
+
+        /// <summary cref="Value.Rebuild(IRBuilder, IRRebuilder)"/>
+        protected internal override Value Rebuild(
+            IRBuilder builder,
+            IRRebuilder rebuilder) =>
+            builder.CreateLanguageVersion(Location);
+
+        /// <summary cref="Value.Accept" />
+        public override void Accept<T>(T visitor) => visitor.Visit(this);
+
+        #endregion
+
+        #region Object
+
+        /// <summary cref="Node.ToPrefixString"/>
+        protected override string ToPrefixString() => "langVersion";
+
+        #endregion
+    }
+
+    /// <summary>
+    /// Represents an inline lanaguage statement.
+    /// </summary>
+    [ValueKind(ValueKind.LanguageEmit)]
+    public sealed class LanguageEmitValue : MemoryValue
+    {
+        #region Instance
+
+        /// <summary>
+        /// Constructs a new inline language statement.
+        /// </summary>
+        /// <param name="initializer">The value initializer.</param>
+        /// <param name="languageKind">The language kind.</param>
+        /// <param name="expressions">The list of all format expressions.</param>
+        /// <param name="hasOutput">Indicates if the first argument is output.</param>
+        /// <param name="arguments">The arguments to format.</param>
+        /// <param name="voidType">The void type.</param>
+        internal LanguageEmitValue(
+            in ValueInitializer initializer,
+            LanguageKind languageKind,
+            FormatArray expressions,
+            bool hasOutput,
+            ref ValueList arguments,
+            VoidType voidType)
+            : base(initializer, voidType)
+        {
+            foreach (var argument in arguments)
```

Comment:
Question: May we put this into `#if DEBUG` to avoid the emission of empty for-loop in release mode?

Reply:
Fixed.